crates/ibc/src/core/ics04_channel/context.rs

```diff
@@ -1,46 +1,28 @@
 //! ICS4 (channel) context.
-
-use crate::core::events::IbcEvent;
-use crate::core::ics02_client::client_state::ClientState;
-use crate::core::ics02_client::ClientExecutionContext;
-use crate::core::ics24_host::path::{
-    ChannelEndPath, ClientConsensusStatePath, CommitmentPath, SeqSendPath,
-};
-use crate::core::{ContextError, ExecutionContext, ValidationContext};
 use crate::prelude::*;
+
 use core::time::Duration;
 use num_traits::float::FloatCore;
 
-use crate::core::ics02_client::consensus_state::ConsensusState;
+use crate::core::events::IbcEvent;
+use crate::core::ics02_client::ClientValidationContext;
 use crate::core::ics03_connection::connection::ConnectionEnd;
 use crate::core::ics04_channel::channel::ChannelEnd;
 use crate::core::ics04_channel::commitment::PacketCommitment;
-use crate::core::ics24_host::identifier::{ClientId, ConnectionId};
+use crate::core::ics24_host::identifier::ConnectionId;
+use crate::core::ics24_host::path::{ChannelEndPath, CommitmentPath, SeqSendPath};
+use crate::core::{ContextError, ExecutionContext, ValidationContext};
 
 use super::packet::Sequence;
 
 /// Methods required in send packet validation, to be implemented by the host
-pub trait SendPacketValidationContext {
-    type ClientValidationContext;
-    type E: ClientExecutionContext;
-    type AnyConsensusState: ConsensusState;
-    type AnyClientState: ClientState<Self::ClientValidationContext, Self::E>;
```

Comment:
In this case, `SendPacketValidationContext` really is a subset of the core `ValidationContext` - which is why we have the blanket implementation if a type already implements `ValidationContext`. I'm not sure if we need `SendPacketValidationContext` really... the `send_packet()` function is not fundamentally different from the rest of the handlers, I think it's safe to assume that everyone will use the same type for `ValidationContext` and `ExecutionContext`.

Reply:
My impression: `SendPacketValidationContext` was defined to provide more specificity and avoid granting excessive access to all `ValidationContext` methods. 
Btw, here is one the places we benefit from having `ClientValidtionContext`